bindata/scripts/load-kmod.sh

```diff
@@ -1,11 +1,13 @@
 #!/bin/sh
 # chroot /host/ modprobe $1
 kmod_name=$(tr "-" "_" <<< $1)
+kmod_args="${@:2}"
 chroot /host/ lsmod | grep "^$1" >& /dev/null
 
 if [ $? -eq 0 ]
 then
+        echo "Module $kmod_name already loaded; no change will be applied..."
```

Comment:
I feel that the vagueness of the error might cause confusion while debugging if the module has already been loaded without enable_unsafe_noiommu_mode=1. I will not know easily that the missing option is the problem. Could it be possible to check that the module is loaded in the way we need it with modinfo and react accordingly (i.e. reload it with the correct options)?

Reply:
yeah honestly I thought about it and I couldn't think of something easy to implement that wasn't too disruptive code-wise. I'll have a second look if you think it's really important.

Reply:
`modinfo vfio` output:

```
filename:       /lib/modules/4.18.0-348.el8.x86_64/kernel/drivers/vfio/vfio.ko.xz
softdep:        post: vfio_iommu_type1 vfio_iommu_spapr_tce
alias:          devname:vfio/vfio
alias:          char-major-10-196
description:    VFIO - User Level meta-driver
author:         Alex Williamson <alex.williamson@redhat.com>
license:        GPL v2
version:        0.3
rhelversion:    8.5
srcversion:     D36061711A57E5FAC95068F
depends:        
intree:         Y
name:           vfio
vermagic:       4.18.0-348.el8.x86_64 SMP mod_unload modversions 
sig_id:         PKCS#7
signer:         Red Hat Enterprise Linux kernel signing key
sig_key:        77:A4:0E:A4:5E:89:11:3E:DC:EE:6E:4B:55:AA:F2:1A:6C:4A:91:21
sig_hashalgo:   sha256
signature:      (...)
parm:           enable_unsafe_noiommu_mode:Enable UNSAFE, no-IOMMU mode.  This mode provides no device isolation, no DMA translation, no host kernel protection, cannot be used for device assignment to virtual machines, requires RAWIO permissions, and will taint the kernel.  If you do not know what this is for, step away. (default: false) (bool)
```

Nothing tells which options are actually loaded.
The only command that I found was `modprobe -c` but it's for all the modules :-/

```
modprobe -c | grep enable_unsafe_noiommu_mode
options vfio enable_unsafe_noiommu_mode=1
```

I find a bit risky to play this game of grepping modprobe and reloading a module if an option wasn't found...

Reply:
Hm, fair point, thank you for checking. Let's then just add a descriptive string that "(assuming module has enable_unsafe_noiommu_mode=1 set)" or something like that, to remind ourselves during the debug process about this.

Reply:
@bn222 I added 2 comments:
https://github.com/k8snetworkplumbingwg/sriov-network-operator/compare/856b088bdd2e4ebd4cb21ff97daac1f29517ad43..37809517fa0fa4ee04e532ef7d935ab3a41b500f

Let me know if it's enough.

Thanks again for your review!

Reply:
ok thanks for taking this into account!